scipy/stats/mstats_basic.py

```diff
@@ -839,10 +839,11 @@ def ttest_1samp(a, popmean, axis=0):
     x = a.mean(axis=axis)
     v = a.var(axis=axis, ddof=1)
     n = a.count(axis=axis)
-    df = n - 1.
-    svar = ((n - 1) * v) / df
-    t = (x - popmean) / ma.sqrt(svar / n)
-    prob = _betai(0.5*df, 0.5, df/(df + t*t))
+    df = n - 1.0
+    with np.errstate(divide='ignore', invalid='ignore'):
+        svar = ((n - 1.0) * v) / df
+        t = (x - popmean) / ma.sqrt(svar / n)
+        prob = special.betainc(0.5*df, 0.5, df/(df + t*t))
```

Comment:
I get a number of warnings of the following type for the fully masked test otherwise. 
`RuntimeWarning: divide by zero encountered in true_divide
probs = special.betainc(0.5*df, 0.5, df/(df + t*t)).reshape(t.shape)`

I agree it's not an elegant solution. If you have a better idea let me know.